### ISSUE TITLE
DCC Validity check shows 14 days waiting time when checking against a booster vaccination certificate

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-documentation#709]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug affects both Android **and** iOS, for specific issues / questions that apply only to one operating system, please raise them in the respective repositories:
    * [Android repository](https://github.com/corona-warn-app/cwa-app-android)
    * [iOS repository](https://github.com/corona-warn-app/cwa-app-ios)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: doesn't matter
- iOS/Android version: doesn't matter
- App version: 2.9.0 (Android) / 2.9.1 iOS

## Describe the bug

The DCC validity check inside of the Corona-Warn-App says that a booster certificate is not yet valid because 14 days haven't passed since the last vaccination. 
While this is technically true, that's not how booster vaccinations work. 
Corona-Warn-App itself does not show a 14 day countdown.
CovPassCheck validates the certificate.

Screenshots:

| Android | iOS |
|---|---|
| ![image](https://user-images.githubusercontent.com/67682506/133728717-d54a64c2-7f9b-4b5f-b278-9789bd8f52de.jpeg) | ![image](https://user-images.githubusercontent.com/67682506/133728767-ab1b6426-bf7e-44a1-b59b-10a71e141c72.jpeg) |

## Steps to reproduce the issue

1. Add a booster vaccination certificate to the Corona-Warn-App ("3/3" for vaccines which required 2 "base" shots, "2/2" for vaccines which required 1 "base" shot.
2. Check the certificates validity by using the "Check validity" feature in CWA

## Expected behaviour

The validity check should not make untrue statements.

**Comment by thomasaugsten** [comment]
The update of the rule is in process. The CovPass should interpret the rule on the same way  and return a warning message

**Comment by Ein-Tim** [comment]
@thomasaugsten 

Yes, CovPass also says that 14 days haven't passed yet, but CovPassCheck says it's valid (that's right). 

Can you give any ETA?

**Comment by thomasaugsten** [comment]
The CovPassCheck is also returning an error with such kind of certificates

**Comment by Ein-Tim** [comment]
@thomasaugsten according to two Twitter users not.

**Comment by thomasaugsten** [comment]
Maybe they can recheck if not please open a issue for covPass Check.
ETA is Monday

**Comment by Ein-Tim** [comment]
@thomasaugsten 

Thanks, I asked them to retest.

**Comment by JonasDeichelmann** [comment]
@thomasaugsten 

The CovPass App says that the certificate is valid and that I have a complete Vaccination protection, see the Screenshot:

![IMG_CCD2CC0215CC-1](https://user-images.githubusercontent.com/20640553/133753418-af96dbaf-43ce-40ce-a8be-63927336d319.jpeg)

**Comment by thomasaugsten** [comment]
This is correct this is also what the CWA is showing. But we talking here about the Gültigkeit prüfen functionality in the both apps

**Comment by JonasDeichelmann** [comment]
Sorry, the "Gültigkeit Prüfen" functionality says even in the CovPass App that it is valid. 

![IMG_65BF078B2A58-1](https://user-images.githubusercontent.com/20640553/133754426-8a7977f9-bae5-4be7-b214-8858349fe1b7.jpeg)

**Comment by thomasaugsten** [comment]
Can you click and tell how many rules are in the check included

**Comment by JonasDeichelmann** [comment]
Sure. What I just recognised it uses the 2/2 certificate even when I have installed the 3/3 certificate for the "Gültigkeit prüfen" functionality.

![IMG_03ECC5A2D421-1](https://user-images.githubusercontent.com/20640553/133758262-f7d58d21-c13b-4d60-b970-e736123e673f.jpeg)

**Comment by thomasaugsten** [comment]
Can you please create a ticket here and explain he is using not the latest certificate for the check
https://github.com/Digitaler-Impfnachweis/covpass-ios/issues

**Comment by dsarkar** [comment]
@Ein-Tim  @JonasDeichelmann Thanks for reporting here.

@JonasDeichelmann Can you confirm that the CovPass**Check** says invalid with a 3/3 certificate (that is currently expected until the "business rules" are updated)

Thanks for your feedback.


---
Corona-Warn-App Open Source Team

**Comment by JonasDeichelmann** [comment]
@dsarkar Yes, I can confirm that the CovPassCheck App says that the 3/3 certificate is invalid. See Screenshot.:
![IMG_8A57AFAB755C-1](https://user-images.githubusercontent.com/20640553/133767043-e220d57d-c576-4bd4-938c-7da8651fd077.jpeg)

**Comment by dsarkar** [comment]
@JonasDeichelmann OK, thanks. Should the ruled be updated (apparently by Monday), both CovPassCheckApp and CWA should show the 3/3 certificate as valid also within the 14 days period. Thanks for the feedback.

**Comment by Ein-Tim** [comment]
@mlenkit Can you give us an update about this issue? Is a fix planned?

**Comment by dsarkar** [comment]
@Ein-Tim I will come back to you. Thanks.

**Comment by mlenkeit** [comment]
@Ein-Tim this should also be covered by the update of the business rule `VR-DE-0003` with `ValidFrom: '2021-10-03T00:00:00Z'` (see also https://github.com/corona-warn-app/cwa-documentation/issues/704#issuecomment-931388095)

**Comment by Ein-Tim** [comment]
Can anyone confirm that this has been fixed? Because then I'd close this issue.

**Comment by jleufgen** [comment]
@Ein-Tim yes the bug was fixed

**Comment by vaubaehn** [comment]
@Ein-Tim I'm not 100% sure (but 80%) that this had been fixed. I tried with a relative's booster vaccination, which took place on 19 Sept. When validating against DE business rules, on/for Sunday 03rd October, it was valid. According to the rule definition, it would take 15 days for any other vaccination (so, validity should have been displayed for Monday October 4th earliest). Looks like it works like intended. 👍
